blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/LocatorFetchRunnable.java

```diff
@@ -155,6 +141,14 @@ public void run() {
         if (log.isDebugEnabled())
             log.debug("Finished {} rollups for (gran,slot,shard) {} in {}", new Object[] {rollCount, parentSlotKey, System.currentTimeMillis() - waitStart});
 
+        finishExecution(waitStart, executionContext);
+    }
+
+    public void waitForRollups() throws InterruptedException {
+        Thread.currentThread().sleep(LOCATOR_WAIT_FOR_ALL_SECS * 1000);
+    }
```

Comment:
Yeah There is nothing going on when InterruptedException gets thrown. What are you really trying to test here? Also waitForRollups just has code to sleep a thread.
            try {
                waitForRollups();
            } catch (InterruptedException ex) {
                if (log.isTraceEnabled())
                    log.trace("Woken wile waiting for rollups to coalesce for {} {}", parentSlotKey);
            } finally {
                String verb = executionContext.doneReading() ? "writing" : "reading";
                log.debug("Still waiting for rollups to finish {} for {} {}", new Object[] {verb, parentSlotKey, System.currentTimeMillis() - waitStart });
            }